### ISSUE TITLE
Http Proxy Preview Feature: A (local?) healthcheck goes to the NVA

### ISSUE THREAD

**Issue by pcallewaert** [Azure/AKS#2815]
**What happened**:
Cluster is configured with the HTTP proxy feature:

```json
"httpProxyConfig": {
      "httpProxy": "http://<redacted>/",
      "httpsProxy": "http://<redacted>/",
      "noProxy": [
        "10.99.0.0/20",
        "10.0.0.0/16",
        "localhost",
        "127.0.0.1",
        "168.63.129.16",
        "169.254.169.254",
        "konnectivity",
        "<redacted>.privatelink.westeurope.azmk8s.io"
      ],
      "trustedCa": null
    },
```

This cluster is configured as a private cluster (with `userDefinedRouting`), with 2 node pools (each node pool has only 1 node). The first node pool is a `system` nodepool, the default node pool.
The second nodepool is a `user` nodepool, nothing is running from my own workloads at this moment.

In the routetable we direct the 0.0.0.0/0 traffic to a transparant HTTP/HTTPS proxy. In the logs of this HTTP proxy server I see 2 requests being denied every second. These requests are not using the configured HTTP_PROXY. (It's the same VMSS, but they would log as `CONNECT` instead of `GET`) This is coming from the node in the default node pool with all the system pods.

```txt
1645799862.810      0 10.99.0.5 TCP_DENIED/403 3862 GET http://:8080/health - HIER_NONE/- text/html
1645799862.857      0 10.99.0.5 TCP_DENIED/403 3862 GET http://:8080/health - HIER_NONE/- text/html
1645799863.810      0 10.99.0.5 TCP_DENIED/403 3862 GET http://:8080/health - HIER_NONE/- text/html
1645799863.855      0 10.99.0.5 TCP_DENIED/403 3862 GET http://:8080/health - HIER_NONE/- text/html
1645799864.809      0 10.99.0.5 TCP_DENIED/403 3862 GET http://:8080/health - HIER_NONE/- text/html
1645799864.855      0 10.99.0.5 TCP_DENIED/403 3862 GET http://:8080/health - HIER_NONE/- text/html
1645799865.810      0 10.99.0.5 TCP_DENIED/403 3862 GET http://:8080/health - HIER_NONE/- text/html
1645799865.856      0 10.99.0.5 TCP_DENIED/403 3862 GET http://:8080/health - HIER_NONE/- text/html
1645799866.810      0 10.99.0.5 TCP_DENIED/403 3862 GET http://:8080/health - HIER_NONE/- text/html
1645799866.855      0 10.99.0.5 TCP_DENIED/403 3862 GET http://:8080/health - HIER_NONE/- text/html
```

I tried to investigate myself, and I was looking into the `coredns` deployment, as this is configured with a liveness to `/health` on port `8080` and 2 replica's, but the interval does not match (is configured for every 10s)

```txt
Liveness:   http-get http://:8080/health delay=60s timeout=5s period=10s #success=1 #failure=5
```

So at this point I don't have an idea from where this is coming. I only know it's coming from the node in the system node pool.

**What you expected to happen**:
It seems these requests are redundant as I think everything is working as expected. The only problem is that this will pollute the logs in the NVA (that is configured in the route table).

**How to reproduce it (as minimally and precisely as possible)**:
I already see this behaviour with a fresh (private) AKS cluster deployed with the default node pool, and configured to use a HTTP proxy, and with userDefinedRouting to a NVA.

**Anything else we need to know?**:

Deployed with terraform

**Environment**:
- Kubernetes version (use `kubectl version`): `1.22.6`
- Size of cluster (how many worker nodes are in the cluster?): 1 in each node pool
- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.): None
- Others:

**Comment by miwithro** [comment]
@alexeldeib

**Comment by pxpnetworks** [comment]
I also have the same unwanted traffic from the AKS cluster going to the Proxy server..

**Comment by alexeldeib** [comment]
we identified an issue with coredns config here and will work on a fix, ETA pending...

**Comment by alexeldeib** [comment]
for posterity this was fixed by https://github.com/coredns/coredns/pull/5401

but AKS doesn't consume the fixed version yet.

cc @robbiezhang

**Comment by dhirschfeld** [comment]
> for posterity this was fixed by [coredns/coredns#5401](https://github.com/coredns/coredns/pull/5401)
> 
> but AKS doesn't consume the fixed version yet (v1.9.4 or greater)
> 
> cc @robbiezhang

It looks like this may have been resolved in https://github.com/Azure/AKS/releases/tag/2023-05-14?

**Comment by alexeldeib** [comment]
I think this should be fixed as mentioned above with recent kubernetes versions? please upgrade and give it a try.

closing for now, feel free to tag if it's still a problem.
